### ISSUE TITLE
Questions about tcn checkpoint parameter count and conversion to PyTorch

### ISSUE THREAD

**Issue by lulucai435** [NVlabs/tiny-cuda-nn#510]
Hi,

I have some questions regarding the TCN model checkpoint:

I converted the MLP part of TCN to PyTorch, but when saving the checkpoint from the original TCN version, I found the total parameter count is 18,432.
According to the network architecture (147 → 64 → 64 → 64 → 4), the expected parameter count should be 18,052.
So, there are about 380 extra parameters — could you please clarify what these extra parameters represent?

Regarding how to convert the TCN checkpoint to the PyTorch format,
is the parameter saving order something like [Layer0_Weights, Layer0_Bias, Layer1_Weights, Layer1_Bias, ...]?
Also, is the memory layout "row-major"?

<img width="1100" height="482" alt="Image" src="https://github.com/user-attachments/assets/e6958aa3-ff39-4214-bb9d-23621bb88b78" />

<img width="906" height="514" alt="Image" src="https://github.com/user-attachments/assets/4b32bce6-33e0-45a2-8628-6892a7734fb4" />

<img width="964" height="230" alt="Image" src="https://github.com/user-attachments/assets/39c62cc7-52ce-4dfd-b8bd-f02ecf7e8170" />

**Comment by Tom94** [comment]
Hi there, [this](https://github.com/NVlabs/tiny-cuda-nn/issues/6#issuecomment-988621068) and [this](https://github.com/NVlabs/tiny-cuda-nn/issues/6#issuecomment-996035736) comment contain all the information.

To summarize:
- row major
- no biases
- input padded to next multiple of 16 with ones
- output padded to next multiple of 16 with values that should be discarded

**Comment by lulucai435** [comment]
Hi again,

Thanks for the previous answers. I have a follow-up question related to the positional encoding in NeRF, specifically regarding the first input dimensions passed to the network.

I’m still unclear whether the positional encoding output includes the original 3 dimensions (x, y, z) or if the encoded vector is just the result of the 24 frequency encodings.
I'm using the NetworkWithInputEncoding class with the following configuration:
`NetworkWithInputEncoding(
  n_input_dims=3, 
  n_output_dims=4, 
  seed=999, 
  dtype=torch.float16, 
  hyperparams={
    'encoding': {'n_frequencies': 24, 'otype': 'Frequency'},
    'network': {
      'activation': 'ReLU', 
      'n_hidden_layers': 3, 
      'n_neurons': 64, 
      'otype': 'FullyFusedMLP', 
      'output_activation': 'None'
    },
    'otype': 'NetworkWithInputEncoding'
  }`
According to my understanding of the positional encoding process in NeRF, each of the 3 input dimensions (x, y, z) will be encoded using 24 frequencies (with sin and cos for each frequency). This results in 3×48=144 dimensions for the encoded input vector.
However, I’m not sure if the original 3 dimensions (x, y, z) are kept alongside the encoded features. So, is the final encoding size 144 or 147 (i.e., 144 + the original 3 dimensions)?

**Comment by Tom94** [comment]
They aren't kept, i.e. the output dimensionality should be 144 in your case. You can find these details [here](https://github.com/NVlabs/tiny-cuda-nn/blob/master/DOCUMENTATION.md)
